### ISSUE TITLE
OBCPBFT (Classic) takes more than 5s to process and finally retrieve query based on TID

### ISSUE THREAD

**Issue by RicHernandez2** [hyperledger-archives/fabric#907]
Description:

OBCPBFT (Classic) cannot retrieve a transaction ID within 5s of the deploy.

Steps to repro:
1.  Setup a pbft classic set of 4 Validating Peers
2.  Deploy via postman setting specific values (b := 20000)
3.  Attempt to do a query every second, for first 5 seconds you will receive the error message captured below.

Results:

Set of 4 pbft classic VPs (not under load) take over 5 seconds to return query with updated value

![query](https://cloud.githubusercontent.com/assets/17052305/14151702/35800468-f67d-11e5-89f6-51f1e1158577.JPG)

Expected Results:

I would expect a faster response from a system doing nothing else, not set to batch mode.

**Comment by gongsu832** [comment]
I have seen the same behavior. I think it's due to the chaincode container being built.

**Comment by genggjh** [comment]
Normally, the container will be built at chaincode deployment stage, here the problem is at query call I think.

Did you check the the status of chaincode container before invoke query API?

**Comment by gongsu832** [comment]
That's what I'm saying. If you send in a query immediately after deploy (which I believe is also an async call like invoke), you will get that error since the container hasn't finished being built yet.

**Comment by RicHernandez2** [comment]
I ran the test this morning to capture the vp0 log (debug) and got some surprising results.  First vp2 requests a view change, then vp3 does as well, then vp0 requests a view change to view 4, which finally succeeds, and makes vp0 the primary again.  It seems something happened overnight to change the primary to something other than vp0 and when I unlocked my laptop this morning, it fought to get the privilege back.  But it didn't do this until my Deploy message came in, I had already opened my laptop, checked my mail, looked at some bugs, only after I read the request for logs here did I go back to repro the issue, and only after the initial deploy did there come a flurry of activity to vote for a new primary, eventually won by vp0's request for view 4.

Here's the logs I took of a deploy/query from this morning, I hit the same problem, it seemed to take even longer than before, probably because they were fighting over who gets to be the primary.

[query.txt](https://github.com/openblockchain/obc-peer/files/197509/query.txt)
